lib/thrift/parser/file_group.ex

```diff
@@ -203,12 +203,15 @@ defmodule Thrift.Parser.FileGroup do
   end
 
   def dest_module(file_group, name) do
-    [thrift_module | struct_name] = name
-    |> Atom.to_string
-    |> String.split(".")
-    |> Enum.map(&String.to_atom/1)
+    name_parts =
+      name
+      |> Atom.to_string
+      |> String.split(".", parts: 2)
 
-    case file_group.ns_mappings[thrift_module] do
+    module_name = name_parts |> Enum.at(0) |> String.to_atom
+    struct_name = name_parts |> Enum.at(1) |> initialcase
+
+    case file_group.ns_mappings[module_name] do
```

Comment:
Could we clean up the rest of the module name creation too because its quite complex compared to what it needs to be.

Without a namespace we can wrap the string in a list to create the alias: `Module.concat([struct_name])`, don't need to add `Elixir`. I think its actually an anti-pattern because `Elixir` is really `:"Elixir.Elixir"`. Fortunately concat will strip all leading `"Elixir."`.

With a namespace we don't need to `Enum.join` or `String.to_atom`. Instead a list of binaries with an alias prepended: `Module.concat([namespace_module | struct_name])`.

Reply:
Those are good suggestions. I've applied them in 7afef2a.

For your last note, I think it needs to be `Module.concat([namespace] ++ [struct_name])`, however, because the namespace is the list and the struct name is the binary.

Reply:
Yes good point. I think I was going by the old data structures for that part.